models/BaseEntity.cfc

```diff
@@ -1165,10 +1183,36 @@ component accessors="true" {
 	 * @return  quick.models.BaseEntity
 	 */
 	private any function loadEntity( required struct data ) {
-		return newEntity()
-			.assignAttributesData( arguments.data )
-			.assignOriginalAttributes( arguments.data )
-			.markLoaded();
+		if (
+			variables._loadChildren
+			&&
+			variables._meta.localMetadata.keyExists( "discriminatorColumn" )
+			&&
+			application.quickMeta.discriminators.keyExists( variables._meta.table )
+			&&
+			data.keyExists( variables._meta.localMetadata.discriminatorColumn )
+			&&
+			application.quickMeta.discriminators[ variables._meta.table ].keyExists(
+				data[ variables._meta.localMetadata.discriminatorColumn ]
+			)
+		) {
+			var childClass = variables._wirebox.getInstance(
+				application.quickMeta.discriminators[ variables._meta.table ][
+					data[ variables._meta.localMetadata.discriminatorColumn ]
+				].mapping
+			);
+
+			keyNames().each( function( key, i ) {
```

Comment:
I've been told lately to avoid `arrayEach` and `structEach` for memory issues in Lucee.  Know anything about that?

Reply:
No, I've heard about massive memory leaks with closures which were `var` scoped so I suppose that might apply to the closure used inside of `each`.  Sounds like something Lucee needs to fix, though.  I've found member function loops to, generally, be faster.